### PR TITLE
Add cmake build

### PR DESCRIPTION
**Pull request by juan42**
Related to issue #24 this PR provides a minimal cmake build configuration. You need to set MFEM_DIR where you have built MFEM (with cmake) so the system can find MFEMConfig.cmake. I rely on pkg-config to find x11, freetype etc. but i think one can safely assume that pkg-config is available on every machine where you want to build glvis.

I'm using this configuration on my machines (Ubuntu 16.04 / CentOS 7) without problems. Preferably this needs to be tested with a CI as well.

One last hint. I'm using the function "convert_filenames_to_full_paths" from the MFEM cmake utils and i copy the whole function. This prevents from having to provide the MFEM source path.

And i just noticed that the installer target is missing.

**TODO**

- [x] Get rid of pkg-config ;)
- [x] Install target

### CODE REVIEW

**Comment by tzanio** [comment]
Hi @juan42, thanks for the contribution!

Do you mind giving us access so we can edit your fork? See https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/.

**Comment by juan42** [comment]
I ticked the box on the right. I always get that wrong, sorry. Next time i will create the repo under the glvis group.

**Comment by trb413** [comment]
I saw your tick box about pkg-config. Pkg-config is often more of a nightmare than its worth, and CMake supports all of the TPLs except fontconfig. So, in my rewrite, everything is native, packaged CMake unless you want Freetype. Then pkg-config still.

It would not be difficult to add FindFontconfig.cmake, but right now, CMakeLists.txt is self-contained, and I get the feeling that @tzanio would prefer to keep it that way! To get rid of pkg-config altogether and still keep it in one file, we could put the logic right in the CMakeLists.txt. After all, it's just looking for libfontconfig; no includes or anything. @tzanio, opinion?

I've got some other things to work on this afternoon, so I'll pick up the install target tomorrow...

**Comment by tzanio** [comment]
I will prefer a simpler CMakeList.txt, but fontconfig is on by default (and nice to have on) :)

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+target_link_libraries(glvis-exe PRIVATE glvis)
+
+# Install the executable
+install(TARGETS glvis-exe RUNTIME DESTINATION bin)
+
+# Install the gnutls helper script
+if (MFEM_USE_GNUTLS)
+  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/glvis-keygen.sh
+    DESTINATION bin
```
Comment (trb413) [inline CMakeLists.txt]: This matches the destination (sort of) from the makefile. The makefile installs things to ```${PREFIX}``` and here we install things to ```${PREFIX}/bin```, which is more consistent with other (e.g., autotools) install paths. Do these really belong together? Perhaps ```${PREFIX}/share```?

**Comment by trb413** [comment]
So, Mac OSX is a bit annoying here. ```find_package(OpenGL)``` will find the OpenGL Framework, which works with Cocoa or whatever but not with XQuartz. Working on a patch.

- [ ] Fix OSX support

**Comment by juan42** [comment]
Seems like i put a lot of work on your back :/. 👍 Works fine on my boxes as well.

**Comment by v-dobrev** [comment]
This branch was started from the `next` branch which is something we want to avoid. New branches should start from the `master` branch.

I will create a new branch `juan42/feature-cmake` and reply all the commits from this branch there.

**Comment by tzanio** [comment]
Superseded by #27.

**Pull request closed without merging**